### ISSUE TITLE
Replace IDXGIAdapter with IDCoreAdapter in gpgmm_d3d12

### ISSUE THREAD

**Issue by mingmingtasd** [intel/GPGMM#683]
We need DXCore to enumerate VPU device, so should we replace IDXGIAdapter with IDCoreAdapter? wdyt? @bjjones @bbernhar @[junwei](https://github.com/fujunwei)

**Comment by bjjones** [comment]
In my brief research it looks like IDXCoreAdapter and IDXGIAdapter are not directly interchangeable. This would require us to use slightly different functions internally for residency management. 

Any comments @bbernhar? Is VPU support in the scope of GPGMM? I'm not sure if there's another reason someone would use DXCore over DXGI.

**Comment by bbernhar** [comment]
I'm supportive and I think it fits with the overall architecture and direction of the project (DXCore backend). @bjjones Can I assign you to dig into this?

**Comment by fujunwei** [comment]
Thanks for support, I think DXGI still is our focus to integrate into Chromium for GPU device, VPU is a longer term plan to support as you changed the title of issue.

**Comment by mingmingtasd** [comment]
> Thanks for support, I think DXGI still is our focus to integrate into Chromium for GPU device, VPU is a longer term plan to support as you changed the title of issue.

Yes, agree. Just propose this here, we should consider supporting DXCore in the near future, maybe both support DXGI/DXCore currently by flag to select (for example, ENABLE_DXCORE= false by default) as [onnx-runtime](https://github.com/microsoft/onnxruntime/blob/main/winml/lib/Api.Image/DeviceHelpers.cpp#L66).
